libtenzir/include/tenzir/contiguous_buffer_stream.hpp

```diff
@@ -0,0 +1,77 @@
+#include "tenzir/fwd.hpp"
+
+#include "arrow/io/file.h"
+
+#include <arrow/array.h>
+#include <arrow/compute/cast.h>
+#include <arrow/io/api.h>
+#include <arrow/table.h>
+
+namespace tenzir {
+
+class contiguous_buffer_stream : public arrow::io::OutputStream {
```

Comment:
For maintainability, we should consider inheriting from BufferOutputStream instead.